zips/zip-0246.rst

```diff
@@ -0,0 +1,566 @@
+::
+
+  ZIP: 246
+  Title: Digests for the Version 6 Transaction Format
+  Owners: Arya <arya@zfnd.org>
+          Conrado Gouvea <conrado@zfnd.org>
+          Daira-Emma Hopwood <daira-emma@electriccoin.co>
+          Jack Grigg <str4d@electriccoin.co>
+          Kris Nuttycombe <kris@electriccoin.co>
+  Status: Draft
+  Category: Consensus
+  Created: 2025-02-12
+  License: MIT
+
+
+===========
+Terminology
+===========
+
+The key words "MUST" and "MUST NOT" in this document are to be interpreted as described
+in BCP 14 [#BCP14]_ when, and only when, they appear in all capitals.
+
+The terms "consensus branch", "epoch", and "network upgrade" in this document are to be
+interpreted as described in ZIP 200. [#zip-0200]_
+
+The term "field encoding" refers to the binary serialized form of a Zcash transaction
+field, as specified in section 7.1 of the Zcash protocol specification
+[#protocol-txnencoding]_.
+
+
+========
+Abstract
+========
+
+This ZIP defines the digest algorithms associated with the v6 transaction format.
+
+This proposal also defines the new concept of "sighash versioning": where previously
+each transaction version had a single associated digest algorithm, going forward it
+will be possible for signers to use any digest algorithm within the closed set
+specified for a given transaction version (and made available in consensus via network
+upgrades).
+
+
+==========
+Motivation
+==========
+
+TBD
+
+TODO: Motivate sighash versioning
+
+
+============
+Requirements
+============
+
+- Continue to support existing functionality of the protocol (multisig,
+  signing modes for transparent inputs).
+
+- It should be possible to update this ZIP with additional digest versions after the first
+  version has been deployed in a network upgrade.
+
+
+================
+Non-requirements
+================
+
+TBD
+
+
+=============
+Specification
+=============
+
+------------------
+Sighash versioning
+------------------
+
+Rough summary:
+
+- Sighash versions are numbered starting from 0 for each tx version.
+- v0 is by convention the "commit to everything" sighash digest. Other versions can commit to whatever makes sense for desired functionality within that tx version.
+- Have a single byte encoded alongside the signature (not appended the way transparent does) that permits the signer to specify which sighash version they are using.
+- Consensus rules choose the digest algorithm for each signer based on that byte.
```

Comment:
We should clarify that the consensus rules choose from the available digest algorithms for the `(consensus_branch_id, transaction_version)` tuple under question (both of which are committed to by the transaction, and can therefore be checked non-contextually). For the NU7 consensus branch ID, there will only be one available digest version for v6 transactions (the v0 digests).